libs/ui/src/lib/datepicker/datepicker-calendar.component.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+  SimpleChanges,
+} from '@angular/core';
+
+@Component({
+  selector: 'cvi-ng-datepicker-calendar',
+  templateUrl: './datepicker-calendar.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DatepickerCalendarComponent implements OnChanges, OnInit {
+  @Input() date?: string;
+  @Input() dayShorthandLabels = ['E', 'T', 'K', 'N', 'R', 'L', 'P'];
+  @Input() monthLabels = [
+    'Jaanuar',
+    'Veebruar',
+    'Märts',
+    'Aprill',
+    'Mai',
+    'Juuni',
+    'Juuli',
+    'August',
+    'September',
+    'Oktoober',
+    'November',
+    'Detsember',
+  ];
+
+  @Output() dateChanged = new EventEmitter<string>();
+
+  today = new Date();
+  selectedDate!: Date;
+  displayDate!: Date;
+  calendarArray: number[][] = [];
+
+  ngOnInit(): void {
+    if (this.calendarArray.length === 0) {
+      this.displayDate = new Date();
+      this.calendarArray = this.getCalendarArray(this.displayDate);
+    }
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes['date'] && this.date) {
+      this.selectedDate = this.parseDate(this.date);
+      this.displayDate = new Date(this.selectedDate);
+      this.calendarArray = this.getCalendarArray(this.displayDate);
+    }
+  }
+
+  parseDate(date: string): Date {
+    const [day, month, year] = date.split('.').map(Number);
+    return new Date(year, month - 1, day);
+  }
+
+  getCalendarArray(date: Date): number[][] {
+    const firstDay = new Date(date.getFullYear(), date.getMonth(), 1).getDay();
+    const adjustedFirstDay = firstDay === 0 ? 7 : firstDay; // adjust for week starting on Monday
+    const numberOfDays = new Date(
+      date.getFullYear(),
+      date.getMonth() + 1,
+      0
+    ).getDate();
+
+    const weeks: number[][] = [];
+    let day = 1;
+    let week: number[] = [];
+
+    // Adjust the first week
+    for (let i = 1; i <= 7; i++) {
+      if (i < adjustedFirstDay) {
+        week.push(0);
+      } else {
+        week.push(day++);
+      }
+    }
+    weeks.push(week);
+
+    // Add the rest of the weeks
+    while (day <= numberOfDays) {
+      week = new Array(7)
+        .fill(0)
+        .map((_, i) => (day <= numberOfDays ? day++ : 0));
+      weeks.push(week);
+    }
+
+    return weeks;
+  }
+
+  getDateClass(day: number, index: number): string {
+    let classes = '';
+
+    const isSelectedDate =
+      this.selectedDate &&
+      day === this.selectedDate.getDate() &&
+      this.displayDate.getMonth() === this.selectedDate.getMonth() &&
+      this.displayDate.getFullYear() === this.selectedDate.getFullYear();
+
+    const isToday =
+      day === this.today.getDate() &&
+      this.displayDate.getMonth() === this.today.getMonth() &&
+      this.displayDate.getFullYear() === this.today.getFullYear();
+
+    if (isSelectedDate) {
+      classes += ' cvi-datepicker__calendar-button-selected';
```

Comment:
All of these classes must be BEM modifiers (inherit the main button class and use an extra `--` name, eg `cvi-datepicker__calendar-button--selected`).